test/cpp_extensions/libtorch_agnostic_extension/libtorch_agnostic/csrc/libtorch_agnostic_kernel.cpp

```diff
@@ -125,3 +125,39 @@ STABLE_TORCH_LIBRARY_IMPL(libtorch_agnostic, CUDA, m) {
 STABLE_TORCH_LIBRARY_IMPL(libtorch_agnostic, CPU, m) {
   m.impl("identity", &boxed_identity);
 }
+
+RAIIATH my_ones_like(RAIIATH t, StableIValue device) {
+  const auto num_args = 6;
+  StableIValue stack[num_args];
+
+  int32_t t_dtype;
+  aoti_torch_get_dtype(t.get(), &t_dtype);
+  auto mf = aoti_torch_memory_format_contiguous_format();
+
+  stack[0] = from(t.release());
+  stack[1] = from(t_dtype); // Dtype
+  stack[2] = from(nullptr); // Layout
+  stack[3] = device; // device
```

Comment:
I didn't figure out how to create a device object from this end haha, so made it an argument 🫣

Reply:
i just removed this code in the end